app/javascript/packs/components/TaskBoard.js

```diff
@@ -1,28 +1,179 @@
-import React from 'react'
-import Board from 'react-trello'
-
-const data = {
-  lanes: [
-    {
-      id: 'lane1',
-      title: 'Planned Tasks',
-      label: '2/2',
-      cards: [
-        {id: 'Card1', title: 'Write Blog', description: 'Can AI make memes', label: '30 mins'},
-        {id: 'Card2', title: 'Pay Rent', description: 'Transfer via NEFT', label: '5 mins', metadata: {sha: 'be312a1'}}
-      ]
+import React, { Component } from 'react';
+import { camelize } from 'humps';
+import Board from 'react-trello';
+import { Button } from 'react-bootstrap';
+import Routes from 'Routes';
+
+import { decamelize } from 'utils/keysConverter';
+import Fetch from './Fetch';
+import LaneHeader from './LaneHeader';
+import AddPopup from './AddPopup';
+import EditPopup from './EditPopup';
+
+export default class TasksBoard extends Component {
+  state = {
+    board: {
+      newTask: null,
+      inDevelopment: null,
+      inQa: null,
+      inCodeReview: null,
+      readyForRelease: null,
+      released: null,
+      archived: null,
     },
-    {
-      id: 'lane2',
-      title: 'Completed',
-      label: '0/0',
-      cards: []
+    addPopupShow: false,
+    editCardId: null,
+  };
+
+  lanesMapping() {
+    return {
+      'newTask': { state: 'new_task', name: 'New', stateEvent: '' },
+      'inDevelopment': { state: 'in_development', name: 'In Dev', stateEvent: 'to_development' },
+      'inQa': { state: 'in_qa', name: 'In QA', stateEvent: 'to_qa' },
+      'inCodeReview': { state: 'in_code_review', name: 'in CR', stateEvent: 'to_code_review' },
+      'readyForRelease': { state: 'ready_for_release', name: 'Ready for release', stateEvent: 'to_ready_for_release' },
+      'released': { state: 'released', name: 'Released', stateEvent: 'release' },
+      'archived': { state: 'archived', name: 'Archived', stateEvent: 'archive' },
     }
-  ]
-}
+  }
+
+  generateLane(id, title) {
+    const tasks = this.state[id];
```

Comment:
You need to create a copy of anything that is being passed by reference like objects, arrays.